src/ddmd/traits.d

```diff
@@ -968,6 +969,109 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         auto se = new StringExp(e.loc, cast(char*)style);
         return se.semantic(sc);
     }
+    if (e.ident == Id.getParameterStorageClasses)
+    {
+        /* Accept a function symbol or a type, followed by a parameter index.
```

Comment:
How about putting this code into a function? This improves readable and avoids making this function even larger (dmd is rather slow optimizing large functions).